src/util/ConfigManager/ValidatorConcept.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (August of 2023, schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <concepts>
+#include <variant>
+
+namespace ad_utility {
+// A validator function is any invocable object, who takes the given parameter
+// types, in the same order and transformed to `const type&`, and returns a
+// bool.
+template <typename Func, typename... ParameterTypes>
+concept Validator =
+    std::regular_invocable<Func, const std::decay_t<ParameterTypes>&...> &&
+    std::same_as<
+        std::invoke_result_t<Func, const std::decay_t<ParameterTypes>&...>,
```

Comment:
Is the `decay_t/` here necessary? You anyway  add a `const &`

Reply:
Good point. changed in [f008ca4](https://github.com/ad-freiburg/qlever/pull/1058/commits/f008ca444b362242e091b044e68a390a5ec0c711).